vaunt-generator/src/main/java/com/hltech/vaunt/generator/domain/representation/RepresentationExtractor.java

```diff
@@ -0,0 +1,53 @@
+package com.hltech.vaunt.generator.domain.representation;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchemaGenerator;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.hltech.vaunt.generator.domain.representation.annotation.Consumer;
+import com.hltech.vaunt.generator.domain.representation.annotation.Provider;
+import com.hltech.vaunt.generator.domain.representation.model.Capabilities;
+import com.hltech.vaunt.generator.domain.representation.model.Contract;
+import com.hltech.vaunt.generator.domain.representation.model.Expectations;
+import com.hltech.vaunt.generator.domain.representation.model.Service;
+import lombok.RequiredArgsConstructor;
+import org.reflections.Reflections;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiredArgsConstructor
+public class RepresentationExtractor {
+
+    private final JsonSchemaGenerator jsonSchemaGenerator;
+
+    public Service extractServiceRepresentation(String packageRoot, String serviceName) throws JsonMappingException {
+        return new Service(serviceName, extractCapabilities(packageRoot), extractExpectations(packageRoot));
+    }
+
+    private Capabilities extractCapabilities(String packageRoot) throws JsonMappingException {
+        List<Contract> providerContracts = new ArrayList<>();
+        for (Class<?> providerMessage : new Reflections(packageRoot).getTypesAnnotatedWith(Provider.class)) {
```

Comment:
why not use streams here?

Reply:
For me in thi particular case current approach is more readable.

Reply:
```java
List<Contract> providerContracts = new Reflections(packageRoot).getTypesAnnotatedWith(Provider.class)
    .stream()
    .map(toContract())
    .collect(Collectors.toList());

return new Capabilities(providerContracts);
```

this is quite nice for readability and "first-look" complexity in my opinion

Reply:
It is not that easy.
1. For now we have no approach to handle exceptions (it must be decided separately)
2. jsonSchemaGenerator.generateSchema(...) throws an exception. It means more complicated implementation than suggested by you -> we would need to handle it in toContract(...) method
3. still I prefer earlier approach

Reply:
Alternative is something similar to:


```suggestion
    private Capabilities extractCapabilities(String packageRoot) {
        List<Contract> providerContracts = new Reflections(packageRoot).getTypesAnnotatedWith(Provider.class).stream()
                .map(this::toContract)
                .collect(Collectors.toList());

        return new Capabilities(providerContracts);
    }

    private Contract toContract(Class<?> providerMessage) {
        JsonSchema body = null;
        try {
            body = jsonSchemaGenerator.generateSchema(providerMessage);
        } catch (JsonMappingException e) {
            e.printStackTrace();
        }
        return new Contract(
                providerMessage.getAnnotation(Provider.class).destinationType(),
                providerMessage.getAnnotation(Provider.class).destinationName(),
                body);
    }
```

Reply:
Alternative would be similar to:

`    private Capabilities extractCapabilities(String packageRoot) {
        List<Contract> providerContracts = new Reflections(packageRoot).getTypesAnnotatedWith(Provider.class).stream()
                .map(this::toContract)
                .collect(Collectors.toList());

        return new Capabilities(providerContracts);
    }

    private Contract toContract(Class<?> providerMessage) {
        JsonSchema body = null;
        try {
            body = jsonSchemaGenerator.generateSchema(providerMessage);
        } catch (JsonMappingException e) {
            e.printStackTrace();
        }
        return new Contract(
                providerMessage.getAnnotation(Provider.class).destinationType(),
                providerMessage.getAnnotation(Provider.class).destinationName(),
                body);
    }`

Reply:
An alternative would be somthing similar to:

`    
private Capabilities extractCapabilities(String packageRoot) {
        List<Contract> providerContracts = new Reflections(packageRoot).getTypesAnnotatedWith(Provider.class).stream()
                .map(this::toContract)
                .collect(Collectors.toList());

        return new Capabilities(providerContracts);
    }

    private Contract toContract(Class<?> providerMessage) {
        JsonSchema body = null;
        try {
            body = jsonSchemaGenerator.generateSchema(providerMessage);
        } catch (JsonMappingException e) {
            e.printStackTrace();
        }
        return new Contract(
                providerMessage.getAnnotation(Provider.class).destinationType(),
                providerMessage.getAnnotation(Provider.class).destinationName(),
                body);
    }
`